bls/src/main/java/tech/pegasys/teku/bls/BLSSecretKey.java

```diff
@@ -13,15 +13,46 @@
 
 package tech.pegasys.teku.bls;
 
+import java.math.BigInteger;
+import java.nio.ByteOrder;
 import java.util.Objects;
 import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.teku.bls.impl.SecretKey;
 
 public final class BLSSecretKey {
 
-  public static BLSSecretKey fromBytes(Bytes32 bytes) {
-    return new BLSSecretKey(BLS.getBlsImpl().secretKeyFromBytes(bytes));
+  private static final Bytes32 CURVE_ORDER =
+      Bytes32.fromHexString("0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001");
```

Comment:
We should probably have a constants class that contains things like this, and `BLS_PUBKEY_SIZE` and other magic numbers. Mikuli extracts them from a Milagro constants class.

The point is that the mechanics of BLS signing are actually somewhat independent of the underlying curve, and it would be cleaner not to let details of the curve leak up to this level. But not a big deal.

Reply:
I moved BLS constants to the `BLSConstants` class, though left SSZ related constants in their corresponding classes. 
I also made a clear distinction between `SSZ` and `Compressed` bytes for `BLSPublicKey` & `BLSSignature` through they are equivalent with the current SSZ implementation
Commits: 25fb8026 and 5dca0433

Does this make sense?

Reply:
:+1: